cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -633,10 +634,15 @@ Result<RecordBatchGenerator> ParquetFileFormat::ScanBatchesAsync(
             kParquetTypeName, options.get(), default_fragment_scan_options));
     int batch_readahead = options->batch_readahead;
     int64_t rows_to_readahead = batch_readahead * options->batch_size;
-    ARROW_ASSIGN_OR_RAISE(auto generator,
-                          reader->GetRecordBatchGenerator(
-                              reader, row_groups, column_projection,
-                              ::arrow::internal::GetCpuThreadPool(), rows_to_readahead));
+    // Modified this to pass the executor in scan_options instead of always using the
+    // default CPU thread pool.
+    // XXX Should we get it from options->fragment_scan_options instead??
```

Comment:
Can we update the comment to remove the `XXX question` and to state what is happening instead of explaining the change? Once is merged the comment won't make much sense as is.
```suggestion
    // Use the executor in scan_options instead of always using the
    // default CPU thread pool.
```